panel/applets/icon-tasklist/Application.vala

```diff
@@ -0,0 +1,283 @@
+public class Application: AttentionStatusListener, IconChangeListener, ScreenProvider{
+
+    private HashTable<unowned Wnck.Window, ApplicationWindow>? appwindow_map = null;
+    // atm, this is the most recently opened window of this application
+    private ApplicationWindow? current_appwin = null;
+
+    private WindowListView? window_list = null;
+    private IconButton button;
+    private unowned WindowManager window_manager;
+    private unowned ApplicationManager app_manager;
+    private uint windows_requiring_attention_num = 0;
+    private GLib.DesktopAppInfo? app_info = null;
+    private unowned Settings settings;
+    private string? wclass_name = null;
+    ulong wclass_id = 0;
+
+
+    public Application (IconButton btn, WindowManager window_manager,
+                        DesktopAppInfo? app_info, Settings settings,
+                        ApplicationManager app_manager){
+
+        appwindow_map = new HashTable<unowned Wnck.Window, ApplicationWindow>(direct_hash, direct_equal);
+        button = btn;
+        this.app_info = app_info;
+        this.settings = settings;
+        this.app_manager = app_manager;
+        button.SetApplication (this);
+
+        this.window_manager = window_manager;
+        window_list = new WindowListView (button, window_manager);
+        button.reset ();
+    }
+
+    public void AddWindow (Wnck.Window window){
+        // create the menu for the new window
+        WindowMenu menu = new WindowMenu (window, app_info, settings, this);
+        ApplicationWindow appwin = new ApplicationWindow (window, menu);
+
+        appwindow_map [window] = appwin;
+        current_appwin = appwin;
+        window_list.AddWindow (appwin);
+
+        // Setup event listeners
+        appwin.SetAttentionStatusListener (this);
+        appwin.SetIconChangeListener (this);
+        // configures the window minimization animation
+        UpdateWindowMinimizeAnimation (window);
+
+        if (GetWindowNum () == 1){
```

Comment:
If understand this correctly, this "C style" coding style should be used for all function names, right?